### ISSUE TITLE
Grove install does sanity checks in earlier commit

### ISSUE THREAD

**Issue by steerio** [panosoft/elm-grove#7]
I forked a package to make it grove installable, for which reason I changed its repository prefix to mine (i.e. in package.json I changed `@origauthor/libname` to `@steerio/libname` and edited the repository field in `elm-package.json` accordingly).

Releasing it went just fine, but trying to install it in a project with `grove install git@github.com:steerio/libname.git` returns the following message:

```
Checking out: steerio/libname (1.0.0)
ERROR: Package name mismatch in elm-package.json in repository field for package: steerio/libname has value of: origauthor/libname.git
```

That is certainly not the case in HEAD (where the tag `1.0.0` also points to), so I looked around in `$TMPDIR` and found a `steerio/libname` folder with the library cloned into. Interestingly, it is an earlier version (namely the commit where I forked the lib) with a folder called `1.0.0` with my release.

Now it appears that the sanity checks done by Grove are being performed on this earlier version as opposed to the release to be installed.

(On a closer look, I cannot decipher the state that this folder is in; `git log -p` shows that HEAD is the point where I forked, while `git status` shows unstaged changes turning my version into that of said HEAD. This looks like an impossible state to me, though it could be that I just don't know git well enough.)

**Comment by cscalfani** [comment]
Did you `bump` the version of the repo before releasing it. The fact that Grove is looking at version 1.0.0 and not your HEAD makes me think that there is no version 1.1.0 or 2.0.0, i.e. there is no NEW version.

Grove works with tags to determine versions not HEAD.

**Comment by steerio** [comment]
1.0.0 is the first and only version. The tag `1.0.0` points to the current HEAD. Grove is in fact looking at an earlier commit, not pointed at by any tag.

Even if this is a bug, it has to be added that normally it's not noticed, as package names don't tend to change. Yet they should be allowed to, and install should validate the current version only. Also, accidentally validating an earlier package state instead of the one to be installed could cause other potential problems.

**Comment by alexgig** [comment]
Hi @steerio 

Are we talking about steerio/elm-web3?

**Comment by steerio** [comment]
Yes. :)

**Comment by alexgig** [comment]
Excellent.

So it looks like the changes you described above (i.e. changing the names etc.) have not been pushed to that repository on github.

Give this a try. Clone your fork. Make the changes locally. Tag the new commit 1.0.1. Push your changes *and* tags. Then try to grove install 1.0.1 from your fork repo.

EDIT: nm, i think i'm starting to see what you did. need another minute to figure out your hist.

EDIT: or, maybe you *just* pushed?

**Comment by steerio** [comment]
I just figured this out myself, too. Pushing explicitly to master seems to have solved the problem. It has nothing to do with Grove, and I'm sorry that I wasted everyone's time here.

To recap, the status exhibiting the problem was that though the tag was there and it *did* point at the proper commit (`6a601`), which was pushed, none of the branches were updated, which must've confused git or Grove.

So the takeaway (to make my foolery somewhat useful for future users) is that `git push --tags` is not enough if the default refspec is not properly set up. To always be safe, use `git push origin branchname --tags`.
